### PR TITLE
chore: add env file validator generator for development

### LINKED ISSUE #8: Add env file validator / generator for development
When the devcontainer boots, a tool should check that the `.env` file exists and that the configuration is properly set. Use an env file parser / validator to ensure that the environment is correct, and generate appropriate default values otherwise.

- How to ensure that the tool and the env vars actually required by strapi code are in sync?

### PR DESCRIPTION
**Pull request by Lumycea**
I've added a python script to add, if necessary, the missing environment variables to the `.env` file (created if needed)

Closes #8

### CODE REVIEW

File: strapi/check_env.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    for line in open("./.env"):
+        match = ENV_REGEX.fullmatch(line)
+        envs[match.group(1)] = match.group(2)
+
+for ee in example_envs.keys():
+    if not ee in envs:
+        print(f"Missing {ee}, adding it with default value \"{example_envs[ee]}\"")
+        envs[ee] = example_envs[ee]
+
+dotenv = open('.env', 'w+')
```
Comment (Robb-Fr) [inline strapi/check_env.py]: What about using this package ? It seems widely adopted https://pypi.org/project/python-dotenv/

Comment (Robb-Fr) [reply strapi/check_env.py]: @Thechi2000

**Review by Robb-Fr — COMMENTED** [review]
All good, just quick comment for @Thechi2000, might be easier, quicker and safer to check the env file

**Review by mochiruntime — COMMENTED** [review]
Wow thank you for your work @Thechi2000! :exploding_head:

There's a few points we could improve imo:

- I don't think we should add another language into the stack. Since we decided to go full typescript for both back and front ends, we should extend this to the entire codebase, including dev scripts. This will reduce the cognitive overhead for the developers, and we can tap into the huge package ecosystem. We should consider using e.g. [ts-node](https://github.com/TypeStrong/ts-node) to run TS scripts directly (it's a common practice to use the [scripts field](https://docs.npmjs.com/cli/v9/using-npm/scripts) of package.json to invoke such utilities).

- As @Robb-Fr mentioned we could use a standard library to load the env vars. [dotenv](https://github.com/motdotla/dotenv) is pretty popular, but maybe we should try finding one that can also perform validation (see my next point).

- I like this idea of using the `.env.example` as reference, however I think we could also check the format of the values themselves at the same time (this will be useful in CI, etc). For instance we could have regexes or even logic for each specific kind of value (e.g. a port should be an int 1-65535, a key a 16-byte base 64 string, etc.). **If we could find a lib for this it would be great (to avoid maintaining this)**. Otherwise we could have a small DSL to express the expected format of `.env` files, for instance:
  ```ts
  // .env-schema.ts
  export const envSchema = {
    HOST: hostname(),
    PORT: port(),
    APP_KEYS: commaSeparated(base64(16), 4),
    // etc...
  }
  ```
  then in the implementation we could have that each DSL call translates to the return of an object with 2 properties: a testing function to check if the supplied value is valid, and a generating function to get a default value. The tool would thus first run in check mode, and then populate missing values in the local `.env` (**which should only happen in dev mode**. In prod, if env are missing the server should just crash since they must be passed as secrets):
  ```ts
  // example implementation for base64()
  import isBase64 from 'is-base-64'
  import crypto from 'crypto'
  export function base64(bytes: number) {
    return {
      isValid: (value: string) => isBase64(value) && value.length === ((bytes + 3 - 1) / 3) * 4,
      defaultValue: () => crypto.randomBytes(bytes).toString('base64'),
    }
  }
  ```
  Otherwise we could try to infer from the `.env.example` file directly, but this sounds like a nightmare :sweat_smile: 

Let me know what you think! Thank you again :blush:

File: strapi/.devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+			]
+		}
+	},
+	// Comment out to connect as root instead. More info: https://aka.ms/vscode-remote/containers/non-root.
+	"remoteUser": "node",
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	// "forwardPorts": [],
+	// Use 'postCreateCommand' to run commands after the container is created.
+	// "postCreateCommand": "yarn install",
+	"postStartCommand": "python3 check_env.py"
```
Comment (mochiruntime) [inline strapi/.devcontainer/devcontainer.json]: I like the idea of starting this through the devcontainer. However, we are limited by VSCode which cannot open multiple devcontainers at the same time, so for instance if someone opens the devcontainer of the NextJS service, this command will not run. I wish we'd have a better, unified scheme, but for now I think our best bet is to use the Dockerfile and inject the command directly: https://github.com/clicepfl/clic-website-v2/blob/10b0dee5aeb23b5be6e265f351904ba1a6a7cc85/strapi/.devcontainer/Dockerfile#L15 
We can use an `entrypoint.sh` script if it gets too crowded (see [entrypoint best practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#entrypoint))

Comment (Lumycea) [reply strapi/.devcontainer/devcontainer.json]: I've looked for librairies in JS, and I've found [joi](https://joi.dev/api/?v=17.7.0) which can both validate and generate values. I'll also use the [dotenv](https://github.com/motdotla/dotenv) as you said.

**Review by Robb-Fr — DISMISSED** [review]
Looks clean to me my bro

**Comment by mochiruntime** [comment]
@Robb-Fr Thanks!
@Thechi2000 Can you also have a look and review through Github? I've changed a few things by adding options and some logs, as well as some comments
I'll squash and merge, however I first want to add some Github Actions to verify the conventional commits spec :blush:

Note: please don't rebase, we've rewritten the files many times

**Comment by Lumycea** [comment]
Good to me

**Review by NoeTerrier — DISMISSED** [review]

**Pull request merged**